lib/util/glob-util.js

```diff
@@ -181,7 +181,11 @@ function listFilesToProcess(globPatterns, options) {
             };
 
             new GlobSync(pattern, globOptions, shouldIgnore).found.forEach(globMatch => {
-                addFile(path.resolve(cwd, globMatch), false, ignoredPaths);
+                const fileMatch = path.resolve(cwd, globMatch);
+
+                if (fs.existsSync(fileMatch) && fs.statSync(fileMatch).isFile()) {
```

Comment:
I'm concerned about the performance impact of doing two additional filesystem accesses for every file that gets linted. I have a few questions:

* Is it necessary to call `fs.statSync` here? Would it be sufficient to just call`fs.existsSync`?
* Presumably, the `file-entry-cache` package is already calling `stat` on the filepath and returning an unexpected type of entry, leading to the crash that you described in https://github.com/eslint/eslint/issues/10039. Would it be possible to just detect that type of entry in `cli-engine.js` rather than doing additional filesystem reads?

Reply:
- `fs.existsSync` returns `false` for a broken symlink, so it seems to be sufficient to just do that.
- The changes I introduced here silently ignore a broken link (no crash with or without the `--cache` option). However, if we only need to fix the crash when using `--cache`, there is indeed a `notFound` property on the cache entry, as well as `err`, which can be rethrown. In that case however, eslint will still crash when there are broken symlinks, but with a more understandable error. But then this doesn't really fix #10039?